test_conformance/SVM/test_unified_svm_free.cpp

```diff
@@ -0,0 +1,182 @@
+//
+// Copyright (c) 2025 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "unified_svm_fixture.h"
+#include "harness/conversions.h"
+#include "harness/testHarness.h"
+#include "harness/typeWrappers.h"
+#include <atomic>
+#include <chrono>
+#include <vector>
+#include <thread>
+
+namespace {
+
+struct CallbackData
+{
+    CallbackData(cl_context ctx): context{ ctx }, status{ 0 }, svm_pointers{} {}
+    cl_context context;
+    std::atomic<cl_uint> status;
+    std::vector<void *> svm_pointers;
+};
+
+// callback which will be passed to clEnqueueSVMFree command
+void CL_CALLBACK callback_svm_free(cl_command_queue queue,
+                                   cl_uint num_svm_pointers,
+                                   void *svm_pointers[], void *user_data)
+{
+    auto data = (CallbackData *)user_data;
+
+    data->svm_pointers.resize(num_svm_pointers, 0);
+
+    for (size_t i = 0; i < num_svm_pointers; ++i)
+    {
+        data->svm_pointers[i] = svm_pointers[i];
+        clSVMFree(data->context, svm_pointers[i]);
```

Comment:
For devices that support system SVM, or in the context of unified SVM when the SVM capabilities includes `CL_SVM_CAPABILITY_SYSTEM_ALLOCATED_KHR`, I think we should be freeing the allocation with the appropriate system free function rather than `clSVMFree` if the SVM pointer was allocated with a system allocation function.